airflow/providers/sftp/sensors/sftp.py

```diff
@@ -21,6 +21,7 @@
 
 import os
 from datetime import datetime, timedelta
+from dateutil.parser import parse as parse_date
```

Comment:
Is it actually required to parse ds/ts which are ISO format representation? Because if a target to parse ISO format (other formats might be not parsed correctly) you could just use a `datetime.datetime.fromisoformat(...)`

Reply:
newer_than accept originaly a datetime type. The meaning of the PR is to accept the same type in a jinja template. 
ds/ts are, you are right in ISO format. 
But if you try this : 
```
from datetime import datetime
from dateutil.parser import parse as parse_date


print(datetime(2020, 1, 2))
print(parse_date("2020-01-02 00:00:00"))
print(parse_date("2018-01-01T00:00:00+00:00"))
print(parse_date("2020-01-02T00:00:00Z"))
```
It will render : 
```
2020-01-02 00:00:00
2020-01-02 00:00:00
2018-01-01 00:00:00+00:00
2020-01-02 00:00:00+00:00

```

Reply:
Note that the same methode is used in the SFTPHook in the deferred mode. I use this method to be coherent with this.

Reply:
Sorry, not the SFTPHook, the SFTPTrigger.

Reply:
> newer_than accept originaly a datetime type. The meaning of the PR is to accept the same type in a jinja template.
ds/ts are, you are right in ISO format. But if you try this :

Don't get this example, could you elaborate a bit more?

> Note that the same methode is used in the SFTPHook in the deferred mode. I use this method to be coherent with this.

This not ever concern if some part already use it, it might be add years ago, or by a mistake.

Reply:
My point here why we need yet another way to parse ISO format, or parse datetime/dates if we already have [`airflow.utils.timezone.parse`](https://github.com/apache/airflow/blob/0a74928894fb57b0160208262ccacad12da23fc7/airflow/utils/timezone.py#L194-L204)

Reply:
Use [airflow.utils.timezone.parse](https://github.com/apache/airflow/blob/0a74928894fb57b0160208262ccacad12da23fc7/airflow/utils/timezone.py#L194-L204) now.